lib/taskana-core/src/main/java/pro/taskana/user/internal/UserServiceImpl.java

```diff
@@ -52,7 +54,13 @@ public User getUser(String id) throws UserNotFoundException {
       throw new UserNotFoundException(id);
     }
 
-    user.setDomains(determineDomains(user.getId()));
+    Set<String> groups =
+        internalTaskanaEngine.executeInDatabaseConnection(() -> userMapper.findGroupsById(id));
+    user.setGroups(Objects.requireNonNullElse(groups, Collections.emptySet()));
+
+    Set<String> accessIds = new HashSet<>(groups);
+    accessIds.add(user.getId());
+    user.setDomains(determineDomains(accessIds.toArray(String[]::new)));
```

Comment:
pass the user object to `determineDomains`?

Reply:
yep

Reply:
makes the call from outside a little cleaner and let's the method do its "dirty" work

Reply:
alright!